### PR TITLE
syncer: move files to new packages to prepare online DDL change

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

At first I found pt and ghost has too many duplications, so I want to add a base type for them. This PR moves some files out of `syncer` package into their own new packages

### What is changed and how it works?

mainly rename

### Check List 

Tests 

 - pass original test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: syncer/metrics/metrics.go [hunk 1]
```diff
@@ -219,76 +213,51 @@ var (
...
-	registry.MustRegister(syncerExitWithErrorCounter)
-	registry.MustRegister(replicationLagGauge)
-	registry.MustRegister(remainingTimeGauge)
-	registry.MustRegister(unsyncedTableGauge)
-	registry.MustRegister(shardLockResolving)
-	registry.MustRegister(heartbeatUpdateErr)
-}
-
-// InitStatusAndMetrics register prometheus metrics and listen for status port.
-func InitStatusAndMetrics(addr string) {
```
Comment (lance6716) [inline syncer/metrics/metrics.go:247]: seems a unused function

File: syncer/dbconn/db.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
+package dbconn
```
Comment (lance6716) [inline syncer/dbconn/db.go:1]: moved from sycner/db.go

File: syncer/metrics/metrics.go (hunk 1 above)
Comment (Ehco1996) [reply syncer/metrics/metrics.go:247]: maybe this used in `dm-syncer` before 😂
I don't know when it was deleted

File: syncer/db_test.go [hunk 3]
```diff
@@ -151,12 +152,12 @@ func (s *testSyncerSuite) TestExecuteSQLSWithIgnore(c *C) {
 	c.Assert(err, IsNil)
 	dbConn, err := db.Conn(context.Background())
 	c.Assert(err, IsNil)
-	conn := &DBConn{
-		baseConn: &conn.BaseConn{
+	conn := &dbconn.DBConn{
```
Comment (lichunzhu) [inline syncer/db_test.go:155]: Should we move these tests into `dbconn` package?

File: syncer/dbconn/db.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+			logCtx.L().Error("fail to close baseDB", log.ShortError(err))
+		}
+	}
+}
+
+// DBConn represents a live DB connection
+// it's not thread-safe.
+type DBConn struct {
+	Cfg      *config.SubTaskConfig
+	BaseConn *conn.BaseConn
```
Comment (lichunzhu) [inline syncer/dbconn/db.go:47]: Why do we need to export these fields?

File: syncer/ddl_test.go [hunk 5]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/pingcap/dm/pkg/log"
 	parserpkg "github.com/pingcap/dm/pkg/parser"
 	"github.com/pingcap/dm/pkg/utils"
+	online_ddl_tools "github.com/pingcap/dm/syncer/online-ddl-tools"
```
Comment (lichunzhu) [inline syncer/ddl_test.go:26]: Doesn't match [Go package names standards](https://blog.golang.org/package-names#TOC_2)
Suggest `onlineddl`.

File: syncer/online-ddl-tools/ghost.go [hunk 6]
```diff
@@ -11,7 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package syncer
+package onlineddl
```
Comment (lichunzhu) [inline syncer/online-ddl-tools/ghost.go:14]: Suggest changing this directory's name to `onlineddl`

File: syncer/online-ddl-tools/online_ddl.go [hunk 7]
```diff
@@ -38,6 +39,11 @@ var OnlineDDLSchemes = map[string]func(*tcontext.Context, *config.SubTaskConfig)
 	config.GHOST: NewGhost,
 }
 
+// refactor to reduce duplicate later.
+var (
+	maxCheckPointTimeout = "1m"
```
Comment (lichunzhu) [inline syncer/online-ddl-tools/online_ddl.go:44]: seems to have a duplicate variable in `syncer/checkpoint.go`?

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: syncer/online-ddl-tools/ghost.go (hunk 6 above)
Comment (lance6716) [reply syncer/online-ddl-tools/ghost.go:14]: there're other folders using `-` like `err-operator`, will rename later 😂

File: syncer/db_test.go (hunk 3 above)
Comment (lance6716) [reply syncer/db_test.go:155]: the logic of `syncer` package is very complex 😂 `ignoreDDLError` belongs to `syncer` now so `syncer/onlineddl` can't access it. We might refactor it in PRs like https://github.com/pingcap/dm/pull/1824

File: syncer/dbconn/db.go (hunk 4 above)
Comment (lance6716) [reply syncer/dbconn/db.go:47]: syncer need to access `BaseConn.DBConn` sometimes

File: syncer/online-ddl-tools/online_ddl.go (hunk 7 above)
Comment (lance6716) [reply syncer/online-ddl-tools/online_ddl.go:44]: Yes, so reduce duplicate later

File: syncer/ddl_test.go (hunk 5 above)
Comment (lance6716) [reply syncer/ddl_test.go]: https://github.com/pingcap/dm/pull/1860/commits/58bb642074b2c0d8b5be52f9259ae77efbf3b289

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**